### PR TITLE
Add new LXC: evcc

### PR DESCRIPTION
**Pull request by MickLesk**
### Add new LXC for evcc
evcc is an extensible EV Charge Controller and home energy management system.

After installation need to run in lxc console:
evcc configure

=> this configure the whole solar, battery and wallboxes in da house

I was able to test it successfully because I have a solar system + storage + wallbox. 
However, the web interface generally always works. ( Shows demo data in this case without configuration)

**Please note! -- Requirements**
1) a supported wallbox or smart power plug
2) a supported measuring device at the house connection, or alternatively a supported PV inverter or other measuring device that measures the current generation power
3) a supported system on which evcc is running (Proxmox LXC is fine)

Optional:
- One or more supported vehicles whose state of charge is retrieved
- further supported wallboxes, switchable sockets, PV generation and battery storage systems
- a supported energy management system (such as SMA Sunny Home Manager) or a dynamic electricity tariff

## Type of change
- [x] New script (a fully functional and thoroughly tested script or set of scripts.)
- [x] Self-review performed (I have reviewed my code, ensuring it follows established patterns and conventions)
- [x] Documentation update required (this change requires an update to the documentation)

image

### CODE REVIEW

**Comment by tteck** [comment]
Why not use their `apt` repository?

**Comment by MickLesk** [comment]
> Why not use their `apt` repository?

Of course we can do that too. I have 2 different scripts. Hold on, I'll push it in.

**Comment by tteck** [comment]
Please don't use their script.

**Comment by MickLesk** [comment]
> Please don't use their script.

i know, the script is wooah :D 🗡️ 

so this is the change to apt repository. ive test it 2 times, intalled good.
image

File: install/evcc-install.sh [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+msg_info "Installing Dependencies"
+$STD apt-get install -y \
+  curl \
+  sudo \
+  mc \
+  debian-keyring \
+  debian-archive-keyring \
+  lsb-release \
+  gpg \
+  apt-transport-https
```
Comment (tteck) [inline install/evcc-install.sh]:
```
debian-keyring 
debian-archive-keyring
apt-transport-https
```

Is not needed

File: install/evcc-install.sh [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+msg_ok "Installed Dependencies"
+
+msg_info "Setting up evcc Repository"
+curl -fsSL https://dl.evcc.io/public/evcc/stable/gpg.EAD5D0E07B0EC0FD.key | gpg --dearmor -o /etc/apt/keyrings/evcc-stable.gpg || msg_error "Failed to download GPG key"
```
Comment (tteck) [inline install/evcc-install.sh]: `|| msg_error "Failed to download GPG key"` not needed
not following established patterns and conventions

File: install/evcc-install.sh [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+echo "deb [signed-by=/etc/apt/keyrings/evcc-stable.gpg] https://dl.evcc.io/public/evcc/stable/deb/debian $(lsb_release -cs) main" >/etc/apt/sources.list.d/evcc-stable.list || msg_error "Failed to add EVCC repository"
```
Comment (tteck) [inline install/evcc-install.sh]: `|| msg_error "Failed to add EVCC repository"` not needed
not following established patterns and conventions

File: install/evcc-install.sh [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+$STD sudo apt update
```
Comment (tteck) [inline install/evcc-install.sh]: `sudo` not needed
not following established patterns and conventions

File: install/evcc-install.sh [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+msg_ok "evcc Repository setup sucessfully"
+
+msg_info "Installing ${APPLICATION}"
```
Comment (tteck) [inline install/evcc-install.sh]: `${APPLICATION}` when vetting the code, you want the app name hard coded.
not following established patterns and conventions

File: install/evcc-install.sh [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+$STD sudo apt install -y evcc
```
Comment (tteck) [inline install/evcc-install.sh]: `sudo` not needed
not following established patterns and conventions

File: install/evcc-install.sh [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+$STD systemctl enable --now evcc.service
```
Comment (tteck) [inline install/evcc-install.sh]: `$STD` not needed
not following established patterns and conventions

File: install/evcc-install.sh [hunk 8]
```diff
@@ -0,0 +1,47 @@
...
+msg_ok "Installed ${APPLICATION}"
```
Comment (tteck) [inline install/evcc-install.sh]: `${APPLICATION}` when vetting the code, you want the app name hard coded.
not following established patterns and conventions

**Comment by tteck** [comment]
I'm a little disappointed that you are still making rookie mistakes. 🤷🏻‍♂️

**Comment by tteck** [comment]
Note for me: `evcc configure`

**Comment by MickLesk** [comment]
The script before was better :D I think I bit my tongue on the sources.list because I had the wrong one for a while and then ignored the rest. Of course it should fit better this way. I'll do a local test

**Comment by MickLesk** [comment]
Test done. Its okay.

**Pull request merged**